internal/discovery/namespace_filter.go

```diff
@@ -0,0 +1,128 @@
+package discovery
+
+import (
+	"errors"
+
+	"github.com/newrelic/nri-kubernetes/v3/internal/config"
+	"github.com/newrelic/nri-kubernetes/v3/internal/storer"
+
+	log "github.com/sirupsen/logrus"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/client-go/informers"
+	"k8s.io/client-go/kubernetes"
+	listersv1 "k8s.io/client-go/listers/core/v1"
+)
+
+// NamespaceFilterer provides an interface to filter namespaces.
+type NamespaceFilterer interface {
+	IsAllowed(namespace string) bool
+}
```

Comment:
I wonder if this interface is too specific. What would be the advantage of having a filter specific to namespaces, as opposed to a filter that takes in `fetch.RawMetrics` and returns a bool? `NamespaceFilter` could be a particular implementation of this, looking for `metrics["namespace"]`

Reply:
You're right, I updated the interface definition and implementation here e57ba695a157903790cc9878ad0f6cbf032beb9f.

Reply:
After talking with @alvarocabanas and @sigilioso, we decided to move back again to the initial definition. Implementing the solution suggested above would imply some refactor in order to avoid cyclic dependencies between definition and discovery packages and to make the code maintainable. 

Done in 95f8b9d068328ecb9f115091592e9d6ab91efe9a. 

We can talk about this when you come back 🙂